backend/dataall/modules/dataset_sharing/services/share_managers/share_manager_utils.py

```diff
@@ -0,0 +1,64 @@
+import abc
+import logging
+
+from dataall.core.environment.db.environment_models import Environment, EnvironmentGroup
+from dataall.modules.dataset_sharing.db.share_object_models import ShareObject
+from dataall.modules.datasets_base.db.dataset_models import Dataset
+
+
+logger = logging.getLogger(__name__)
+
+
+class ShareManagerUtils:
+    def __init__(
+            self,
+            session,
+            dataset: Dataset,
+            share: ShareObject,
+            source_environment: Environment,
+            target_environment: Environment,
+            source_env_group: EnvironmentGroup,
+            env_group: EnvironmentGroup,
+    ):
+        self.target_requester_IAMRoleName = share.principalIAMRoleName
+        self.session = session
+        self.dataset = dataset
+        self.share = share
+        self.source_environment = source_environment
+        self.target_environment = target_environment
+        self.source_env_group = source_env_group
+        self.env_group = env_group
+
+    def add_missing_resources_to_policy_statement(
+            self,
+            resource_type,
+            target_resources,
+            existing_policy_statement,
+            iam_role_policy_name
+    ):
+        """
+        Checks if the bucket and kms resources is in the existing policy. Otherwise, it will add it.
```

Comment:
as the function is written in a general way, I'd remove "bucket" and "kms" from the description

Reply:
Makes sense! Removed it in the next commit.